daal4py/sklearn/ensemble/GBTDAAL.py

```diff
@@ -196,15 +207,18 @@ def fit(self, X, y):
     def _predict(
         self, X, resultsToEvaluate, pred_contribs=False, pred_interactions=False
     ):
-        # Input validation
-        if not self.allow_nan_:
-            X = check_array(X, dtype=[np.single, np.double])
-        else:
-            X = check_array(X, dtype=[np.single, np.double], force_all_finite="allow-nan")
-
         # Check is fit had been called
         check_is_fitted(self, ["n_features_in_", "n_classes_"])
 
+        # Input validation
+        X = validate_data(
+            self,
+            X,
+            dtype=[np.single, np.double],
```

Comment:
```suggestion
            dtype=[np.float64, np.float32],
```
Order matters here for conversion, unless we always want to default to ```single```.  Also might as well follow convention elsewhere.

Reply:
Fixed in all places for these estimators.